### ISSUE TITLE
bitonic

### ISSUE THREAD

**Issue by inferrna** [pyopencl/pyopencl#79]
Is there any way to sort signed float values with pyopencl? For implemented radix sort keys must be "integer-valued C expressions", but bitonic sort haven't this limitation.

**Comment by inducer** [comment]
No, not currently, but I'd love a patch!

**Comment by inferrna** [comment]
I just have finished bitonic sorter
https://gist.github.com/inferrna/2202fb8938fbc63e54dc
It also can sort multidimensional array, like numpy does, but with one
limitation - sorted dimension size must be power of 2. I requested Eric
Bainville for license of his work and he answered that all code published
on bealto.com is under BSD license. So I also added some copyrights, to
register it.

2015-06-24 0:02 GMT+08:00 Andreas Klöckner <notifications@github.com>:
